### PR TITLE
Add Umbrella directories

### PR DESCRIPTION
**Pull request by hu5970**
Add umbrella directories based on PR from RRFSv1.
Also, Added options in ungrib to handle operation GFS and GEFS grib2 files.

Tested umbrella directories on JET with real-time MPAS runs.

### CODE REVIEW

File: workflow/rocoto_funcs/base.py
Comment (guoqing-noaa) [inline workflow/rocoto_funcs/base.py:53]: what does `wgf` stand for?

File: jobs/JRRFS_FCST
Comment (guoqing-noaa) [inline jobs/JRRFS_FCST:34]: 1. It is my impression that variables defined in a J-job and to be exported start with the uppercase letters.   I would suggest changing to:
`export UMBRELLA_DATA=${UMBRELLA_DATA:-${DATAROOT}}`

2. `UNBRELLA_DATA`  -> `UMBRELLA_DATA`

3. It looks like `umbrella_data` is a replicate of `DATAROOT` if not defined at the job card/workflow level. This may cause misunderstanding when users read the scripts.  
Suggest changing to:
`export UMBRELLA_DATA=${UMBRELLA_DATA:-UMBRELLA_DATA_not_defined}`
or
`export UMBRELLA_DATA=${UMBRELLA_DATA:-${DATAROOT}/umbrella}`

File: workflow/rocoto_funcs/base.py
Comment (guoqing-noaa) [inline workflow/rocoto_funcs/base.py:81]: `UNBRELLA_DATA` to `UMBRELLA_DATA` and other locations

File: jobs/JRRFS_FCST
Comment (hu5970) [reply jobs/JRRFS_FCST:34]: Done

File: workflow/rocoto_funcs/base.py
Comment (guoqing-noaa) [reply workflow/rocoto_funcs/base.py:53]: Thanks, @BenjaminBlake-NOAA !

@hu5970 I would suggest we add a comment on line 53 or other appropriate locations
`wgf=os.getenv('WGF','det')  # WGF: Working Group Function, det, enkf, enfs, firewx`

File: workflow/rocoto_funcs/base.py
Comment (hu5970) [reply workflow/rocoto_funcs/base.py]: Done

File: scripts/exrrfs_ungrib.sh
Comment (hu5970) [reply scripts/exrrfs_ungrib.sh:50]: I will simplify those later when I can run a test with those changes.

File: workflow/rocoto_funcs/base.py
Comment (hu5970) [reply workflow/rocoto_funcs/base.py:53]: I forgot to commit changes for "exp/exp.jet_conus3km". I added this explanation for WGF in that configure file.

File: workflow/exp/exp.jet_conus3km
Comment (guoqing-noaa) [inline workflow/exp/exp.jet_conus3km:6]: @hu5970 Thanks for adding the comment!

@BenjaminBlake-NOAA @MatthewPyle-NOAA 
I don't know whether we have a chance to change `ensf` as it looks very similar to `enkf` or `ensrf` (variants of ensemble Kalman filters)
Changing to something like `enfs` may reduce some confusion. Just a random thought.

Comment (BenjaminBlake-NOAA) [reply workflow/exp/exp.jet_conus3km:6]: @guoqing-noaa  Let's see if @lgannoaa has any comments on changing ensf because he is the one who added WGF to the workflow for RRFSv1.  If we were to change it, my vote would be for enfcst, which is slightly longer but makes it clear you are referring to the ensemble forecast component.

Comment (lgannoaa) [reply workflow/exp/exp.jet_conus3km:6]: I will recommend not to change the ensf for now, so we have clear reference. @guoqing-noaa

Comment (guoqing-noaa) [reply workflow/exp/exp.jet_conus3km:6]: @BenjaminBlake-NOAA @lgannoaa 
Thanks for the input!

If we can use `enfcst`, that will be sweeter! I can imagine `ensf` may cause lots of confusion when more people use the mpas+jedi-based rrfs-workflow. I would suggest we make this change for the `rrfs-mpas-jedi` branch as early as possible.

**Comment by hu5970** [comment]
Hi All, we can use the WGF discussion to later.  Thanks, Ming

File: scripts/exrrfs_ungrib.sh
Comment (guoqing-noaa) [reply scripts/exrrfs_ungrib.sh:50]: > I will simplify those later when I can run a test with those changes.

Do you mean to make changes in another PR or make changes after testing but still in this PR?

File: scripts/exrrfs_mpassit.sh
Comment (MatthewPyle-NOAA) [inline scripts/exrrfs_mpassit.sh:1]: Could the comment at line 41 be updated to reflect that the data is moved to $UMBRELLA_DATA rather than copied to COMOUT?

**Review by MatthewPyle-NOAA — COMMENTED** [review]
One small request, but pretty much ready to approve/merge once @guoqing-noaa approves.

File: scripts/exrrfs_ungrib.sh
Comment (hu5970) [reply scripts/exrrfs_ungrib.sh:50]: I will make it in another PR, right after this one is merged.

Comment (guoqing-noaa) [reply scripts/exrrfs_ungrib.sh:50]: Looks good to me. Thanks!

**Review by guoqing-noaa — APPROVED** [review]

File: scripts/exrrfs_mpassit.sh
Comment (hu5970) [reply scripts/exrrfs_mpassit.sh:1]: Changed

**Review by MatthewPyle-NOAA — APPROVED** [review]

File: scripts/exrrfs_mpassit.sh
Comment (guoqing-noaa) [reply scripts/exrrfs_mpassit.sh:1]: Thank @MatthewPyle-NOAA for spotting this. Yes, I agree that it is better to update the outdated comments.

File: workflow/exp/exp.jet_conus3km
Comment (guoqing-noaa) [reply workflow/exp/exp.jet_conus3km:6]: For reference, `ENSF` does mean one kind of filer, i.e. `Ensemble Score Filter`

**Pull request merged**
